### ISSUE TITLE
Press dial on streamdeck+ not working ? 

### ISSUE THREAD

**Issue by FabXxXxXx** [pre-martin/StreamDeckSimHubPlugin#97]
Hello,

I've just discovered your wonderful plugin (tahnk you !), so I begin my journey in mapping :)
I have tried to map the control of the Road Rumble Effect via control mapper. The CW (Volume +) and CCW (Volume -) are working flawlessly. But when I press the dial (mute button) to map the action on the Bass Shaler module, nothing's happened.
I have tried with "Simhub control" or "Simhub Role" but it's not working in either case.
Do you know what I've done wrong ? 

![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/0d50480d-fb0a-4fc8-87bf-467ea6a91e95)

**Comment by pre-martin** [comment]
"Roles" are primarily for the input controller abstraction in SimHub (if you have lots of different wheels and you don't want to switch controller profiles in your simulation). So you should use the "SimHub Control" field.

The documentation contains a complete example. Did you already take a look at https://github.com/pre-martin/StreamDeckSimHubPlugin/blob/main/doc/shakeit/ShakeIt.adoc ?

**Comment by FabXxXxXx** [comment]
Yes, and thank you to have taking the time to write a full documentation by the way :)

I have made some tests : everything is working, using both "Simhub Roles" or "Simhub Control" if I use a button, or a dial CW or CCW. But when using Dial Press, it doesn't work at all. I don't understand why.

For reference :
- Simhub v 9.2.9
- Streamdeck v 6.6.0 (20583)
- Stream Deck SimHub Plugin v 1.10.5 ( 3 dec 2023)

What I've tested so far :

![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/e571db83-6a11-4050-b42c-c4af9f7d4649)

![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/ee4e384e-cd09-43e2-8939-76486fec8cf3)

![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/e807cd45-1ed4-4bfc-9f9c-7cd31e7b1a85)

![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/5d6c98b9-84eb-41d0-9beb-6426789b9343)

**Comment by FabXxXxXx** [comment]
I have found the log of the Stream Deck SimHub Plugin. This is what's inside. Maybe something has changed in the last versions of streamdeck software ? 

![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/b2271eaf-bb23-4c5b-98b8-29e5f24b25d2)

**Comment by pre-martin** [comment]
I am using a library for my Stream Deck plugin. Stream Deck already announced that they will remove some events. And until a few days ago, the library I am using did not switch to the new events. Let me check this tomorrow....

**Comment by FabXxXxXx** [comment]
Hi, 

I have modified and compiled a new dll from Sharpdeck, because it's not working with the 7.0.0 beta 1. Now the dial up and dial down is transmitted :

![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/8d0271a6-251b-4fc7-aa89-106753323368)

But I can't make it work inside your plugin. I have modified : 

- DialAction.cs :
![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/80382fdb-6c44-41f0-a799-d38260cbad59)

- Action.cs :
![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/28b01ccf-56c2-48d0-bfb3-fffefbe66065)

- DialActionSettings.cs :
![image](https://github.com/pre-martin/StreamDeckSimHubPlugin/assets/168359978/6acdbb90-d7be-4e0c-ba54-8814f9ff8ab5)

But no doubt that is something else elsewhere :)

I put the new Sharpdeck.dll if you want to try before a new release.
[SharpDeck.zip](https://github.com/pre-martin/StreamDeckSimHubPlugin/files/15182716/SharpDeck.zip)

I'm really far from being a developper, i'm more of a kind of "copy-paster". So my apologises for everything I have done wrong with the code :)

**Comment by pre-martin** [comment]
Thank you for this detailed analysis.

I created a ticket in the upstream project (https://github.com/GeekyEggo/SharpDeck/issues/22), but I am not sure if this patch will be applied. To project seems to be dead.

So I forked SharpDeck and applied the patch into my local repository. With these changes I was able to use the dial button again.

I will release a new version of my plugin either tomorrow or on Saturday.
